lib/rubocop/cop/style/random_with_offset.rb

```diff
@@ -0,0 +1,179 @@
+          add_offense(node)
+        end
+
+        private
+
+        def autocorrect(node)
+          lambda do |corrector|
+            if integer_op_rand?(node)
+              corrector.replace(node.source_range,
+                                corrected_integer_op_rand(node))
+            elsif rand_op_integer?(node)
+              corrector.replace(node.source_range,
+                                corrected_rand_op_integer(node))
+            elsif rand_modified?(node)
+              corrector.replace(node.source_range,
+                                corrected_rand_modified(node))
+            end
+          end
+        end
+
+        def corrected_integer_op_rand(node)
+          left, operator, right = node.children
+
+          offset = int_from_int_node(left)
+
+          prefix_node, _, random_node = right.children
+
+          prefix = prefix_from_prefix_node(prefix_node)
+          left_int, right_int = boundaries_from_random_node(random_node)
+
+          if operator == :+
+            "#{prefix}(#{offset + left_int}..#{offset + right_int})"
+          else
+            "#{prefix}(#{offset - right_int}..#{offset - left_int})"
+          end
+        end
+
+        def corrected_rand_op_integer(node)
+          left, operator, right = node.children
+
+          prefix_node, _, random_node = left.children
+
+          offset = int_from_int_node(right)
+
+          prefix = prefix_from_prefix_node(prefix_node)
+          left_int, right_int = boundaries_from_random_node(random_node)
+
+          if operator == :+
+            "#{prefix}(#{left_int + offset}..#{right_int + offset})"
+          else
+            "#{prefix}(#{left_int - offset}..#{right_int - offset})"
+          end
+        end
+
+        def corrected_rand_modified(node)
+          rand, method = node.children
+          prefix_node, _, random_node = rand.children
+
+          prefix = prefix_from_prefix_node(prefix_node)
+          left_int, right_int = boundaries_from_random_node(random_node)
+
+          if %i[succ next].include?(method)
+            "#{prefix}(#{left_int + 1}..#{right_int + 1})"
+          elsif method == :pred
+            "#{prefix}(#{left_int - 1}..#{right_int - 1})"
+          end
+        end
+
+        def prefix_from_prefix_node(node)
+          if node.nil?
+            'rand'
+          else
+            _, prefix = node.children
+            "#{prefix}.rand"
+          end
+        end
+
+        def boundaries_from_random_node(random_node)
+          children = random_node.children
+
+          case random_node.type
+          when :int
+            [0, int_from_int_node(random_node) - 1]
+          when :irange
+            [int_from_int_node(children.first),
+             int_from_int_node(children[1])]
+          when :erange
+            [int_from_int_node(children.first),
+             int_from_int_node(children[1]) - 1]
+          end
+        end
+
+        def int_from_int_node(node)
+          node.children.first
```

Comment:
Probably we need some richer interface for value node types to avoid adding similar code in numerous places.

Reply:
I remember there's a `str_content` method in node, maybe we can have something similar?

Reply:
@bbatsov: It's quite high on my list of priorities.